### PR TITLE
Improve error detection for zypper_call and add_suseconnect_product

### PR DESCRIPTION
**Pull request by foursixnine**
While working on adding AI/ML to SLE15, I came across these two problems:

* `add_suseconnect_product` was not failing if the call to `SUSEConnect` failed, hence giving us a false positive, here's how it looks now: http://phobos.suse.de/tests/3856979#step/tflite2/33
* `zypper_call` now will report package conflicts detected http://phobos.suse.de/tests/3856984#step/tflite2/34

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -500,12 +500,20 @@ sub zypper_call {
     $IN_ZYPPER_CALL = 1;
     # Retrying workarounds
     my $ret;
+    my $search_conflicts = 'awk \'BEGIN {print "Processing Record - ",NR; group=0}
```
Comment (foursixnine) [inline lib/utils.pm:503]: @rwx788 @SergioAtSUSE @okurz I can't recall if we have code that will already detect such package conflicts (at least I could not find any)

**Comment by foursixnine** [comment]
> I cannot check for integrity of the AWK part, but the rest and the results of the PR LGTM.

It works, just look at the referenced test, also acn be tested locally :)... in any case, i'll wait a bit more in case an awk wizard has more knowledge on how to get better `blocks`, I'll automerge by EOD if nobody else has objections anyhow

File: lib/registration.pm [hunk 2]
```diff
@@ -189,6 +189,9 @@ sub add_suseconnect_product {
             record_soft_failure 'bsc#1124318 - Fail to get module repo metadata - running the command again as a workaround';
         }
         assert_script_run("SUSEConnect -p $name/$version/$arch $params", $timeout);
+    } elsif ($result && !$retry) {
+        die "SUSEConnect failed activating module $name with exit code (see log output): $result.";
+
```
Comment (rwx788) [inline lib/registration.pm:194]: Nitpicking, let's remove extra empty line =)

Comment (foursixnine) [inline lib/registration.pm:194]:
```suggestion
```

File: lib/utils.pm [hunk 3]
```diff
@@ -500,12 +500,21 @@ sub zypper_call {
...
```
Comment (foursixnine) [inline lib/utils.pm:503]:
```suggestion
    my $search_conflicts = 'awk \'BEGIN {print "Processing conflicts - ",NR; group=0}
```

**Pull request merged**